src/parsec_client/operations/psa_algorithm.md

```diff
@@ -147,6 +144,11 @@ Asymmetric signature algorithms. Supported algorithms:
    defined by [RFC 6979](https://tools.ietf.org/html/rfc6979.html). Uses one of the hash algorithm
    supported.
 
+When defining the permitted algorithms in a key policy, the hash-and-sign algorithms above can use
+the value **Any Hash** for their hash algorithm, meaning that it will allow any hash algorithm. This
+value must not be used to build an algorithm specification to perform an operation. It is only valid
+to build policies.
```

Comment:
I totally agree with this, but I'd like to enforce it somewhere before the providers as well - in the interface or client, maybe. Not compulsory, just thinking out loud here

Reply:
Aren't we doing that with the `SignHash` type?

Reply:
I meant the fact that policy algorithms support wildcards, but operation algorithms don't, we don't enforce that do we?

Reply:
I think we do because `Any` has been removed from [`Hash`](https://docs.rs/parsec-interface/0.15.0/parsec_interface/operations/psa_algorithm/enum.Hash.html)

Reply:
? [`PsaSignHash`](https://github.com/parallaxsecond/parsec-interface-rs/blob/master/src/operations/psa_sign_hash.rs#L18) takes an `AsymmetricSignature` algorithm. [`AsymmetricSignature`](https://github.com/parallaxsecond/rust-psa-crypto/blob/master/psa-crypto/src/types/algorithm.rs#L317) accepts `SignHash` as the hash algorithm. So there's nothing preventing a client using `Any` for `PsaSignHash`

Reply:
oh absolutely, my bad 😅